backend/rust-gbt/src/audit_transaction.rs

```diff
@@ -50,19 +51,24 @@ impl PartialEq for AuditTransaction {
 impl Eq for AuditTransaction {}
 
 #[inline]
-pub fn partial_cmp_uid_score(a: (u32, f64), b: (u32, f64)) -> Option<Ordering> {
+pub fn partial_cmp_uid_score(a: (u32, u32, f64), b: (u32, u32, f64)) -> Option<Ordering> {
     // If either score is NaN, this is false,
     // and partial_cmp will return None
-    if a.1 == b.1 {
-        Some(a.0.cmp(&b.0))
+    if a.2 != b.2 {
+        a.2.partial_cmp(&b.2)
+    } else if a.1 != b.1 {
+        Some(b.1.cmp(&a.1))
```

Comment:
This is also flipped.

Maybe there should be a comment in both places about the flip.

(Also, the only reason I didn't use this fn in TxPriority was because the self and other were flipped. Maybe we can use this fn by flipping the inputs somehow, that way this is the only place we need to comment.)

Reply:
good point, it would be nice to consolidate that logic into a single ordering